src/modm/driver/temperature/max31865.hpp

```diff
@@ -0,0 +1,249 @@
+// coding: utf-8
+// ----------------------------------------------------------------------------
+/*
+ * Copyright (c) 2023, Henrik Hose
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_MAX31865_HPP
+#define MODM_MAX31865_HPP
+
+#include <array>
+#include <modm/architecture/interface/register.hpp>
+#include <modm/architecture/interface/spi_device.hpp>
+#include <modm/processing/resumable.hpp>
+#include <modm/processing/timer.hpp>
+
+namespace modm
+{
+
+struct Pt
+{
+	float R0;
+	float Rref;
+	float alpha;
+	double a;
+	double b;
+	double c;
+};
+
+/// Standard values for IEC 751 (PT100) with 430 Ohm reference
+constexpr Pt pt100{
+	.R0 = 100.f,
+	.Rref = 430.f,
+	.alpha = 3.85055e-3f,
+	.a = double{3.90830e-3},
+	.b = double{-5.77500e-7},
+	.c = double{-4.18301e-12},
+};
+
+/// @ingroup modm_driver_max31865
+// Standard values for IEC 751 (PT100) with 400 Ohm reference
+template<Pt pt>
+struct max31865
+{
+
+	enum class Fault : uint8_t
+	{
+		RtdHighThreshold = Bit7,
+		RtdLowThreshold = Bit6,
+		RefinMGreater85Percentbias = Bit5,
+		RefinMLess85PercentVbias = Bit4,
+		RtdinMLess85PercentVbias = Bit3,
+		OverUnderVoltage = Bit2
+	};
+	MODM_FLAGS8(Fault)
+
+	enum class Config : uint8_t
+	{
+		VBias = Bit7,
+		ConversionModeAuto = Bit6,
+		OneShot = Bit5,
+		ThreeWire = Bit4,
+		FaultDetectionCycle1 = Bit3,
+		FaultDetectionCycle0 = Bit2,
+		FaultStatusClear = Bit1,
+		Rejection = Bit0
+	};
+	MODM_FLAGS8(Config)
+
+	enum class FaultDetectionWrite : uint8_t
+	{
+		NoAction = 0b00,
+		FaultDetectionAutomaticDelay = 0b01,
+		RunFaultDetectionWithManualDelayCycle1 = 0b10,
+		FinishFaultDetectionWithManualDelayCycle2 = 0b11,
+	};
+	using FaultDetectionWrite_t =
+		Configuration<Config_t, FaultDetectionWrite, 0b11, 2>;  // Bit 8..10
+
+	enum class FaultDetectionRead : uint8_t
+	{
+		FaultDetectionFinished = 0b00,
+		AutomaticDetectionStillRunning = 0b01,
+		ManualCycle1Running =
+			0b10,  // waiting for user to write FinishFaultDetectionWithManualDelayCycle2
+		ManualCycle2StillRunning = 0b11,
+	};
+	using FaultDetectionRead_t = Configuration<Config_t, FaultDetectionRead, 0b11, 2>;  // Bit 8..10
+
+	enum class Rejection : uint8_t
+	{
+		Rejection60Hz = 0b0,
+		Rejection50Hz = 0b1,
+	};
+	using Rejection_t = Configuration<Config_t, Rejection, 0b1>;  // Bit 8..10
+
+	struct FaultThreshold
+	{
+		using RtdResistance_t = uint16_t;
+	};
+
+	static constexpr uint16_t
+	rtdfaultthresh(const FaultThreshold::RtdResistance_t res)
+	{
+		return static_cast<uint16_t>(res << 1);
+	};
```

Comment:
What is this used for?
Is the `FaultThreshold::RtdResistance_t` typedef really necessary or could one just pass an `uint16_t` to `rtdfaultthresh( )`?